File: python/Warmup-2/string_match.py
def string_match(a, b):
    shortest = min(len(a), len(b))
    count = 0

    # use shorter substring to determine equality
    for i in range(shortest-1):
        a_sub = a[i:i+2]
        b_sub = b[i:i+2]
        if a_sub == b_sub:
            count += 1

    return count

File: java/Warmup-2/stringMatch.java
public int stringMatch(String a, String b) {
  int shortest = Math.min(a.length(), b.length()),
      count = 0;
  
  for(int i = 0; i < shortest - 1; i++) {
    if(a.substring(i, i+2).equals(b.substring(i, i+2))) count++;
  }
  return count;
}
